Set "FluentLocale" cookie only when user has accepted the necessary cookies

moritz-sauer-13 opened the pull request.

xini commented:
Thanks for this, @moritz-sauer-13.

Can you please change the namespace from `Innoweb\CookieConsent\Injectors` to `Innoweb\CookieConsent\Middleware` and the class to `DetectLocaleWithCookieConsentMiddleware`?

Also, I would argue that the fluent cookie is not strictly necessary and should be in the category `CookieConsent::PREFERENCES`?

moritz-sauer-13 commented:
Thanks for your input, @xini.

I've updated the namespace to Innoweb\CookieConsent\Middleware and renamed the class to DetectLocaleWithCookieConsentMiddleware as suggested.

I also agree with your point regarding the Fluent cookie – it’s not strictly necessary and fits better in the CookieConsent::PREFERENCES category. I’ve updated that as well.

Let me know if anything else needs adjusting.

xini commented:
Thanks, and sorry for the delay. This is now tagged as [3.10.0](https://github.com/xini/silverstripe-cookie-consent/releases/tag/3.10.0).

xini commented:
Hm.. that's interesting. I'll have a look.

xini commented:
This should now be fixed in 3.10.1. I have added a check whether the fluent class exists before creating the middleware. I think it was the psr-4 class loader that loads the class tree, even if the classes are not actually used.
